### PR TITLE
README.md updated, small edits to content and manifest (#5)

### PR DESCRIPTION
**Pull request by hamletv**

### CODE REVIEW

File: content.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const myPort = browser.runtime.connect({name:"content-port"}); //connection to b
 
 myPort.onMessage.addListener((message) => {  // listener for message from background script
 
-  let images = document.querySelectorAll(`img[src="${message.srcUrl}"]`);  //if image repeats in page, alt attr will be changed for all.
+  let images = document.querySelectorAll(`img[src="${message.srcUrl}"]`);  //if image repeats in page, text will populate in all images.
```
Comment (yzen) [inline content.js]: nit: lets put this commend about the variable declaration (on line 4)

File: README.md [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+# Image A11y Extension
 
-latest source files xpi
-how to use Extension
-info about dependencies
-screenshots of using Extension
+Image A11y is an accessibility extension that allows screen reader users to click on an image and obtain the text content of that image.
+It is not a description of the image but the actual text content within the image that was previously inaccessible to screen reader users. Think of memes, comics, diagrams and any photos with superimposed text on them as part of the image, this is content non-sighted users are unable to access.
+
+![Screenshot of headline, blurb of Firefox news](/Users/villa/Desktop/image-a11y-screenshots/Screenshot_2018-07-27 New Firefox Releases Now Available.png)
```
Comment (yzen) [inline README.md]: Path here is relative to your machine. It should be uploaded somewhere public and then referenced via URL

File: README.md [hunk 3]
```diff
@@ -1,6 +1,31 @@
...
+
+Typically the text in this image would not be accessible to a screen reader and therefore to its user. Image A11y allows you to obtain the image text on demand by right clicking on the image to be scanned and passing the text into the reassigned alt attribute.
+
+Image scanning is done with the use of [Tesseract.js](https://github.com/naptha/tesseract.js), a JavaScript library that has been around for several years and initially developed by HP. The extension works by establishing a communication port between the background and content scripts. Tesseract and the scanning it performs works in the background script, which has a connection with the content script that serves to post the message containing the scanning results and the srcUrl of the image. Once the message (components) from the background script are in the content script they are passed into the DOM, specifically the scanned text is passed into the alt attribute of the image scanned, making it now accessible to the screen reader.
```
Comment (yzen) [inline README.md]: I think we can drop the description of how it works (e.g. background/content communication description). We can eventually add more comments in files themselves.

File: README.md [hunk 4]
```diff
@@ -1,6 +1,31 @@
...
+
+Scanning accuracy is not 100%. There are multiple variables within the image itself that diminish the accuracy of the scanning, poorest accuracy is seen with memes and images that contain significant "noise" around the text portion of the image. Best results, highest accuracy is seen with images or screenshots of text; images with the text on a white background and similar cases where the text within the image is unobstructed. Training of tesseract is possible to increase accuracy and this will be the next step in making this extension more helpful to the user.     
+
+## How to use Extension
+
+![Screenshot of image with text, right click drop down menu](/Users/villa/Desktop/image-a11y-screenshots/Screen Shot 2018-07-27 at 9.06.07 AM.png)
```
Comment (yzen) [inline README.md]: Same here, path here is relative to your machine. It should be uploaded somewhere public and then referenced via URL

File: README.md [hunk 5]
```diff
@@ -1,6 +1,31 @@
...
+Right click on the image you want to scan. The Image A11y icon and "Discover Image Text" option will show at the bottom of the menu once extension has been added and activated.
```
Comment (yzen) [inline README.md]: nit: remove added and activated part. I'd add at the start of the sentence "Once extension is installed, right click ..."

Comment (yzen) [reply README.md]: change menu to context menu

File: README.md [hunk 6]
```diff
@@ -1,6 +1,31 @@
...
+
+![Screenshot of image with text, inspector open showing image attributes](/Users/villa/Desktop/image-a11y-screenshots/Screen Shot 2018-07-27 at 9.06.45 AM.png)
```
Comment (yzen) [inline README.md]: Same here, path here is relative to your machine. It should be uploaded somewhere public and then referenced via URL

Comment (yzen) [reply README.md]: The screen should should not have the alt text at all. ```alt=""``` is a valid case that means "presentation only"

File: README.md [hunk 7]
```diff
@@ -1,6 +1,31 @@
...
+Prior to scanning and clicking "Discover Image Text" you can see the img alt attribute is empty. The attribute will be reassigned and scanned text will be injected there.
+
+![Screenshot of image with text, inspector open, clicking on Image A11y icon](/Users/villa/Desktop/image-a11y-screenshots/Screen Shot 2018-07-27 at 9.07.36 AM.png)
```
Comment (yzen) [inline README.md]: Same here, path here is relative to your machine. It should be uploaded somewhere public and then referenced via URL

File: README.md [hunk 8]
```diff
@@ -1,6 +1,31 @@
...
+With inspector still open (which is not needed during regular use), alt attribute still empty, right click and "Discover Image Text".
+
+![Screenshot of image with text, inspector open, Image A11y clicked](/Users/villa/Desktop/image-a11y-screenshots/Screen Shot 2018-07-27 at 9.08.08 AM.png)
```
Comment (yzen) [inline README.md]: Same here, path here is relative to your machine. It should be uploaded somewhere public and then referenced via URL

File: README.md [hunk 9]
```diff
@@ -1,6 +1,31 @@
...
+Image has been scanned and the alt attribute now has text from image. Alt attribute text always starts with "Possible text:" due to the potential inaccuracies mentioned above.
+
+
+## Dependencies
+
+[Tesseract](https://github.com/naptha/tesseract.js)
```
Comment (yzen) [inline README.md]: Lets also include the actual version (that we are using) of tesseract here too.

File: package.json [hunk 10]
```diff
@@ -17,6 +17,7 @@
   },
   "homepage": "https://github.com/hamletv/image_a11y_extension#readme",
   "dependencies": {
-    "tesseract.js": "~1.0.10"
+    "tesseract.js": "~1.0.10",
+    "web-ext": "^2.8.0"
```
Comment (yzen) [inline package.json]: web-ext should go inside devDependencies property.

Comment (hamletv) [reply package.json]: Thanks for that yzen!

**Comment by yzen** [comment]
Hi @hamletv , I don't think this pull request was ready to be merged just yet. There was 1 more change I requested (web-ext dependency in the package.json). I'll open the follow up.

**Pull request merged**